src/extension/layer-filters/src/main/java/au/org/emii/geoserver/extensions/filters/LayerFiltersService.java

```diff
@@ -148,6 +153,11 @@ public void uniqueValues(HttpServletRequest request, HttpServletResponse respons
         String layer = request.getParameter("layer");
         String propertyName = request.getParameter("propertyName");
 
+        if(workspace==null) {
```

Comment:
Is it worth a slight change to avoid running that split method twice? e.g.
```java
        if(workspace==null) {
            String[] splitLayerArray = splitLayerName(layer)
            workspace = splitLayerArray[0];
            layer = splitLayerArray[1];
        }
```